### PR TITLE
feat: add SettleDealPayments method and types

### PR DESCRIPTION
**Pull request by LexLuthr**
Fixes https://github.com/filecoin-project/go-state-types/issues/239

### CODE REVIEW

File: builtin/v13/market/market_types.go [hunk 1]
```diff
@@ -138,3 +139,31 @@ type GetDealActivationReturn struct {
 type GetDealSectorParams = DealQueryParams
 
 type GetDealSectorReturn = abi.SectorNumber
+
+type SettleDealPaymentsParams struct {
+	Deals bitfield.BitField
```
Comment (Stebalien) [inline builtin/v13/market/market_types.go:144]: Should be:

```go
type SettleDealPaymentParams = bitfield.BitField
```

(the upstream type "transparently" encodes as a bitfield without wrapping it in a struct)

File: builtin/v13/market/market_types.go [hunk 2]
```diff
@@ -138,3 +139,31 @@ type GetDealActivationReturn struct {
...
+}
+
+type SettleDealPaymentsReturnResultFailCodes struct {
```
Comment (Stebalien) [inline builtin/v13/market/market_types.go:147]: You can just call this a `FailCode`.

Comment (Stebalien) [reply builtin/v13/market/market_types.go:147]: (we may want to move it to a shared location later anyways)

File: builtin/v13/market/market_types.go [hunk 3]
```diff
@@ -138,3 +139,31 @@ type GetDealActivationReturn struct {
...
+	Index    uint64
+	ExitCode exitcode.ExitCode
+}
+
+type SettleDealPaymentsReturnResult struct {
```
Comment (Stebalien) [inline builtin/v13/market/market_types.go:152]: `BatchReturn`

File: builtin/v13/market/market_types.go (hunk 2 above)
Comment (Stebalien) [reply builtin/v13/market/market_types.go:147]: Yeah, we'll move both `FailCode` and `BatchReturn` to `util` in a followup patch (removing it from the verifreg.

**Comment by LexLuthr** [comment]
@Stebalien I have updated the names as per the suggestions. I have not defined `market.SettleDealPaymentsParams{}` in gen.go since it is unwrapped and bitfield.Bitfield already has Marshal and Unmarshal methods.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**